internal/worker/fortress/occupy.go

```diff
@@ -28,14 +28,23 @@ func Occupy(ctx context.Context, fortress Guest, start StartFunc) (worker.Worker
 	// creation; and a worker-running func that sends on exactly one
 	// of them, and returns only when (1) a value has been sent and (2)
 	// no worker is running. Note especially that it always returns nil.
-	started := make(chan worker.Worker, 1)
-	failed := make(chan error, 1)
+	started := make(chan worker.Worker)
+	failed := make(chan error)
 	task := func() error {
 		worker, err := start()
 		if err != nil {
-			failed <- err
+			select {
+			case failed <- err:
+			case <-ctx.Done():
+			}
 		} else {
-			started <- worker
+			select {
+			case started <- worker:
+			case <-ctx.Done():
+				// If a worker is a little slow to start, it may occupy
+				// indefinitely if the context passed here is cancelled.
+				worker.Kill()
```

Comment:
We should mention that it's ok to kill the work multiple times, and because the context is done, we want to be sure that worker is dead and not orphaned.